logbus/solvermon/solvermon.go

```diff
@@ -74,29 +74,48 @@ func (sm *SolverMonitor) handleBuildkitStatus(ctx context.Context, status *clien
 	for _, vertex := range status.Vertexes {
 		meta, operation := vertexmeta.ParseFromVertexPrefix(vertex.Name)
 		var cmdID string
+		createCmd := true
 		switch {
 		case meta.TargetName == "context":
 			cmdID = operation
 		case meta.CommandID != "":
-			cmdID = fmt.Sprintf("%s/%s", meta.TargetID, meta.CommandID)
+			cmdID = meta.CommandID
+			createCmd = false
 		default:
 			cmdID = vertex.Digest.String()
 		}
 		vm, exists := sm.vertices[cmdID]
 		if exists {
 			sm.digests[vertex.Digest] = cmdID
 		} else {
-			// TODO(vladaionescu): Should logbus commands be created in the converter instead?
 			category := meta.TargetName
 			if meta.Internal {
 				category = fmt.Sprintf("internal %s", category)
 			}
-			cp, err := bp.NewCommand(
-				cmdID, operation, meta.TargetID, category, meta.Platform,
-				vertex.Cached, meta.Local, meta.Interactive, meta.SourceLocation,
-				meta.RepoGitURL, meta.RepoGitHash, meta.RepoFileRelToRepo)
-			if err != nil {
-				return err
+			var cp *logbus.Command
+			// Operations initiated from Earthly have created Logbus commands
+			// ahead-of-time. Others may originate from BuildKit, so we'll have
+			// to create a command at this point.
+			if createCmd {
+				var err error
+				cp, err = bp.NewCommand(
```

Comment:
Oh.. so this is still necessary for things that don't map directly to Earthfile commands, right? Stuff like "exporting outputs", "context", and other internal operations, right?

Reply:
Yep, you got it. I wasn't able to remove this completely as some statuses don't map to commands on the other end.

Reply:
Yeah, makes sense...